Integrations/MISP_V2/MISP_V2.py

```diff
@@ -743,6 +743,58 @@ def check_url():
     else:
         demisto.results(f'No events found in MISP for URL: {url}')
 
+def build_misp_complex_filter(demisto_query: str):
+    """
+    Args:
+        demisto_query: complex query contains saved words: 'AND:', 'OR:' and 'NOT:'
+            using ',' as delimiter for parameters and ';' as delimiter for operators.
+            using the operators is optional.
+            if 'demisto_query' does not contains any of the complex operators the original
+            input will be returned
+
+    Returns:
+        dict: dictionary created for misp to perform complex auery
+        or if no complex qury found retruns the original input
+
+    Example:
+        demisto_query should look like:
+            example 1: "AND:param1,param2;OR:param3;NOT:param4,param5"
+            example 2: "NOT:param3,param5"
+            example 3 (simple syntax): "param1,param2"
+    """
+
+    regexAnd = r"(AND:)([^\;]+)(;)"
+    regexOr = r"(OR:)([^\;]+)(;)"
+    regexNot = r"(NOT:)([^\;]+)(;)"
+    andList = None
```

Comment:
it is best practise to initialize a dict, and then assign it in the conditions, and then user the `**` operator to pass it to `MISP.build_complex_query` instead of initializing variables with `None`

Reply:
done